db/src/urls.rs

```diff
@@ -139,7 +149,26 @@ async fn store_markdown(
     Ok(markdown_result)
 }
 
-async fn store_raw_html_in_page_snapshot(
+pub async fn generate_and_store_summary(
```

Comment:
Added a generate summary article function to call later in `store_html_content_in_page_snapshot`